parsers/test/test_ES.py

```diff
@@ -1,216 +1,188 @@
-            ZoneKey("ES-IB-MA"),
-            self.session,
-            datetime.fromisoformat("2023-09-03"),
-        )
-
-        # Test "get" function has been called correctly
-        self.assertEqual(
-            mocked_session_get.call_args[0][0],
-            "https://demanda.ree.es/WSvisionaMovilesBalearesRest/resources/demandaGeneracionBaleares?curva=MALLORCA5M&fecha=2023-09-03",
-        )
-
-        # Test that the data is parsed correctly afterwards
-        self.assertEqual(len(data_list), 1)
-        self.assertEqual(data_list[0]["sortedZoneKeys"], "ES->ES-IB-MA")
-        self.assertEqual(data_list[0]["source"], "demanda.ree.es")
-        self.assertIsNotNone(data_list[0]["netFlow"])
-        self.assertEqual(data_list[0]["netFlow"], 309.8)
-        self.assertTrue(isinstance(data_list[0]["datetime"], datetime))
-        self.assertEqual(
-            data_list[0]["datetime"],
-            datetime(2023, 9, 3, 17, 5, tzinfo=timezone.utc),
-        )
-
-    # Test for DST change days
-    @patch("requests.Response")
-    @patch("parsers.ES.Session.get")
-    def test_production_DST_IB(self, mocked_session_get, mocked_response):
-        mocked_response.ok = True
-        mocked_response.text = r'null({"valoresHorariosGeneracion":[{"ts":"2020-10-25 2A:55","dem":261.3,"car":73.3,"die":0.0,"gas":0.0,"cc":127.1,"cb":100.4,"fot":0.0,"tnr":0.0,"trn":0.0,"eol":0.0,"emm":-3.5,"emi":-58.1,"otrRen":0.0,"resid":19.2,"genAux":0.0,"cogen":2.9,"eif":-0.0},{"ts":"2020-10-25 2B:00","dem":261.7,"car":73.3,"die":0.0,"gas":0.0,"cc":128.0,"cb":100.4,"fot":0.0,"tnr":0.0,"trn":0.0,"eol":0.0,"emm":-3.3,"emi":-58.1,"otrRen":0.0,"resid":18.5,"genAux":0.0,"cogen":2.9,"eif":-0.0}]}'
-        mocked_session_get.return_value = mocked_response
-        data_list = ES.fetch_production(
-            ZoneKey("ES-IB-MA"), self.session, datetime.fromisoformat("2020-10-25")
-        )
-
-        # Test "get" function has been called correctly
-        self.assertEqual(
-            mocked_session_get.call_args[0][0],
-            "https://demanda.ree.es/WSvisionaMovilesBalearesRest/resources/demandaGeneracionBaleares?curva=MALLORCA5M&fecha=2020-10-25",
-        )
-
-        # Test that the data is parsed correctly afterwards
-        self.assertEqual(len(data_list), 2)
-        self.assertTrue(isinstance(data_list[0]["datetime"], datetime))
-        self.assertEqual(
-            data_list[0]["datetime"].astimezone(timezone.utc),
-            datetime(2020, 10, 25, 0, 55, tzinfo=timezone.utc),
-        )
-        self.assertTrue(isinstance(data_list[1]["datetime"], datetime))
-        self.assertEqual(
-            data_list[1]["datetime"].astimezone(timezone.utc),
-            datetime(2020, 10, 25, 1, 0, tzinfo=timezone.utc),
-        )
-
-
-if __name__ == "__main__":
-    main()
+### El Hierro
+# Consumption
+@patch("requests.Response")
+@patch("parsers.ES.Session.get")
+def test_fetch_consumption(mocked_session_get, mocked_response, session):
```

Comment:
This seems to still work but do you think using unittest.mock for this will cause issues in the future?

Reply:
No i think it will be fine, its just a way to manipulate a functions response and such I think, independent on how a test is executed etc, but I think it can be done also by using the request adapter as i saw in most other tests.

But, its a difference to how its done elsewhere, and being systematic is often positive from a maintenance perspective long term. I didnt feel very confident to make a call or dig into this so it seemed reasonable to not change as part of this

Reply:
I'm happy to try migrate this to look like most other test using adapter fixture, but i figure its better to avoid bundling it in this oversized PR

Reply:
Let's keep it as is for now then!